### ISSUE TITLE
Dependency conflicts

### ISSUE THREAD

**Issue by ploeh** [fsprojects/Paket#2383]
### Description

I'm not quite sure if this is a _paket_ issue, or an _FsCheck_ issue, but I chose to report it here because I _can't_ repro the issue with Visual Studio's NuGet package manager console.

When adding `FsCheck.Xunit`, Visual Studio gives a compiler warning about conflicts in dependencies I don't even understand why they're there.

I'm not sure if the issue is related to `FsCheck.Xunit` at all, or if it's wider, but so far, I've only seen it in connection to `FsCheck.Xunit`.

### Repro steps

1. Create a new F# library project using Visual Studio 2015.
1. Create a `.paket` directory: `$ mkdir .paket`
1. Copy `paket.bootstrapper.exe` into `.paket`.
1. Run the bootstrapper: `$ .paket/paket.bootstrapper.exe`
1. Initialise _paket:_ `$ .paket/paket init`
1. Add `FsCheck.Xunit` to the project: `$ .paket/paket add nuget FsCheck.Xunit project ConflictRepro`
1. From within Visual Studio, _rebuild all_.

### Expected behavior

The project should compile without errors or warnings. This is also the case if, instead of using _paket_, you use the _Package Manager Console_ built into Visual Studio. Adding `FsCheck.Xunit` via the _Package Manager Console_ does _not_ cause the issue.

### Actual behavior

The project compiles, but with a warning:

```
Severity	Code	Description	Project	File	Line	Suppression State
Warning		Found conflicts between different versions of the same dependent assembly that could not be resolved.  These reference conflicts are listed in the build log when log verbosity is set to detailed.	ConflictRepro	C:\Program Files (x86)\MSBuild\14.0\bin\Microsoft.Common.CurrentVersion.targets	1820	
```

Turning on diagnostics logging, the relevant information seems to be this:

```
There was a conflict between "System.Security.Cryptography.Algorithms, Version=4.1.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" and "System.Security.Cryptography.Algorithms, Version=4.2.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a".
    "System.Security.Cryptography.Algorithms, Version=4.1.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" was chosen because it was primary and "System.Security.Cryptography.Algorithms, Version=4.2.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" was not.
    References which depend on "System.Security.Cryptography.Algorithms, Version=4.1.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" [C:\Users\mark\Desktop\ConflictRepro\packages\System.Security.Cryptography.Algorithms\lib\net461\System.Security.Cryptography.Algorithms.dll].
        C:\Users\mark\Desktop\ConflictRepro\packages\System.Security.Cryptography.Algorithms\lib\net461\System.Security.Cryptography.Algorithms.dll
          Project file item includes which caused reference "C:\Users\mark\Desktop\ConflictRepro\packages\System.Security.Cryptography.Algorithms\lib\net461\System.Security.Cryptography.Algorithms.dll".
            System.Security.Cryptography.Algorithms
    References which depend on "System.Security.Cryptography.Algorithms, Version=4.2.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" [].
        C:\Users\mark\Desktop\ConflictRepro\packages\System.Security.Cryptography.OpenSsl\lib\netstandard1.6\System.Security.Cryptography.OpenSsl.dll
          Project file item includes which caused reference "C:\Users\mark\Desktop\ConflictRepro\packages\System.Security.Cryptography.OpenSsl\lib\netstandard1.6\System.Security.Cryptography.OpenSsl.dll".
            System.Security.Cryptography.OpenSsl
```

It seems that _paket_ adds all those `System.Security.Cryptography.*` assemblies as references to the project, even though they don't seem to be required. The _Package Manager Console_ doesn't add those references.

### Known workarounds

Remove the `System.Security.Cryptography.*` references.

This isn't the best workaround, however, because every time I ask _paket_ to update packages, or do other work, it re-adds those references.

**Comment by forki** [comment]
I tried to reproduce, but Paket 5 beta behaves correct. No warnings shown.

[Library1.zip](https://github.com/fsprojects/Paket/files/1044268/Library1.zip)

Please retry with Paket 5 beta. (BTW: in the zip you can just copy the .paket folder and paket magic mode is enabled with prerelease channel)

**Comment by ploeh** [comment]
I agree that the library you've posted doesn't have the issue, but I can still reproduce the issue with the above steps and [Paket 5.0.0-rc001](https://github.com/fsprojects/Paket/releases/tag/5.0.0-rc001).

One difference that I've noted is that the library you've posted is an F# 3.1 library, whereas the one I'm creating is an F# 4.1 library.

I tried going through the steps with an F# 3.1 library, but somehow, it turns into an F# 4.1 library along the way...

**Comment by forki** [comment]
> One difference that I've noted is that the library you've posted is an F# 3.1 library, whereas the one I'm creating is an F# 4.1 library.

can you please upload a zip with the repro? I'm not really sure what I'm doing differently

**Comment by ploeh** [comment]
> can you please upload a zip with the repro?

I'd love to, but GitHub will not let me upload an 85 MB .zip file. Should I remove the `packages` directory?

**Comment by forki** [comment]
yes packages dir can be restored (given that it's a public url)

2017-06-06 14:29 GMT+02:00 Mark Seemann <notifications@github.com>:

**Comment by ploeh** [comment]
Here we go, then: [Library2.zip](https://github.com/fsprojects/Paket/files/1055014/Library2.zip)

**Comment by forki** [comment]
I can't repro. I don't see any warnings in the build.
@matthid can you please try to repro?

**Comment by ploeh** [comment]
It could be environment-specific (_doesn't work on my machine_). I'll see if I can reproduce on my other machine...

**Comment by ploeh** [comment]
The plot thickens...

I tried downloading the [above](#issuecomment-306473057) Library2.zip to another machine. The same problem also manifests on that other machine.

Then I tried going through the repro steps I originally described. On that other machine, I _can't_ reproduce the issue from first principles...
